Api/Services/EventService.cs

```diff
@@ -603,5 +607,76 @@ public async Task<Result> DeleteEventAsync(int eventId, User user)
 
             return Result.Success;
         }
+
+        /// <summary>
+        /// Returns events with specified optional parameters
+        /// </summary>
+        /// <param name="request">parameters of the request</param>
+        /// <returns>list of events that fulfill the requirements</returns>
+        public async Task<Result<List<EventVM>>> GetEventsAsync(GetEventsRequest request)
+        {
+            var events = context.Events.Where(e => e.EventId == e.EventId);
+            if (request.Name is not null)
+            {
+                events = events.Where(e => e.Name.Contains(request.Name.Trim()));
+            }
+            if (request.DateFrom is not null)
+            {
+                events = events.Where(e => e.Time > request.DateFrom);
+            }
+            if (request.DateUntil is not null)
+            {
+                events = events.Where(e => e.Time < request.DateUntil);
+            }
+            if (request.EventStatus is not null)
+            {
+                switch (request.EventStatus)
+                {
+                    case EventStatus.Future:
+                        events = events.Where(e => e.MustJoinUntil > DateTime.UtcNow);
+                        break;
+                    case EventStatus.Past:
+                        events = events.Where(e => e.Time < DateTime.UtcNow);
+                        break;
+                    default:
+                        events = events.Where(e => (e.MustJoinUntil < DateTime.UtcNow && e.Time > DateTime.UtcNow) || e.MaxPeople == e.ParticipationRequests.Count);
```

Comment:
```suggestion
                        events = events.Where(e => (e.MustJoinUntil < DateTime.UtcNow || e.MaxPeople == e.ParticipationRequests.Count) && e.Time > DateTime.UtcNow);
```
mi się wydaje że nie chcemy tu zwracać przeszłych wydarzeń